### PR TITLE
Adding GitHub action that compares and fails if one of the current bundle sizes grows more then 10KB

### PR DESCRIPTION
**Pull request by barboaron**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,10 @@
     "createVersion": "node scripts/createVersion",
     "storybook": "npm run start --prefix examples/storybook",
     "exampleApp": "npm run start --prefix examples/main",
-    "analyzeBundles": "npm run build && npm run analyzeBundles:nobuild",
+    "saveBundles": "npm run build && npm run save --prefix plugins-bundle-analyzer",
+    "minify": "uglifyjs --some --options",
+    "analyzeBundlesAndCompare": "npm run analyzeBundles --silent && npm run compare --prefix plugins-bundle-analyzer --silent && npm run minify --silent",
```
Comment (shaulgo) [inline package.json:29]: maybe analyzeBundles should always compare. Why not?

File: package.json [hunk 2]
```diff
@@ -24,7 +24,10 @@
     "createVersion": "node scripts/createVersion",
     "storybook": "npm run start --prefix examples/storybook",
     "exampleApp": "npm run start --prefix examples/main",
-    "analyzeBundles": "npm run build && npm run analyzeBundles:nobuild",
+    "saveBundles": "npm run build && npm run save --prefix plugins-bundle-analyzer",
+    "minify": "uglifyjs --some --options",
```
Comment (shaulgo) [inline package.json:28]: what's this for?

File: package.json [hunk 3]
```diff
@@ -24,7 +24,10 @@
     "createVersion": "node scripts/createVersion",
     "storybook": "npm run start --prefix examples/storybook",
     "exampleApp": "npm run start --prefix examples/main",
-    "analyzeBundles": "npm run build && npm run analyzeBundles:nobuild",
+    "saveBundles": "npm run build && npm run save --prefix plugins-bundle-analyzer",
```
Comment (shaulgo) [inline package.json:27]: rename to saveBundlesSizesBaseline

File: plugins-bundle-analyzer/savingBundlesSizes.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"wix-rich-content-plugin-button":226,"wix-rich-content-plugin-code-block":221,"wix-rich-content-plugin-divider":210,"wix-rich-content-plugin-emoji":180,"wix-rich-content-plugin-file-upload":212,"wix-rich-content-plugin-gallery":817,"wix-rich-content-plugin-giphy":212,"wix-rich-content-plugin-hashtag":214,"wix-rich-content-plugin-headers-markdown":178,"wix-rich-content-plugin-html":216,"wix-rich-content-plugin-image":216,"wix-rich-content-plugin-line-spacing":181,"wix-rich-content-plugin-link-preview":219,"wix-rich-content-plugin-link":207,"wix-rich-content-plugin-map":335,"wix-rich-content-plugin-mentions":209,"wix-rich-content-plugin-sound-cloud":292,"wix-rich-content-plugin-text-color":208,"wix-rich-content-plugin-undo-redo":178,"wix-rich-content-plugin-video":297,"viewer-without-wrapper":198,"viewer-with-wrapper":217,"editor-with-emoji":925,"editor-without-plugins":836,"editor-with-basic-plugins":1277}
```
Comment (shaulgo) [inline plugins-bundle-analyzer/savingBundlesSizes.json:1]: rename file to bundlesSizesBaseline

File: plugins-bundle-analyzer/compareBundles.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+const chalk = require('chalk');
+const fs = require('fs');
+
+function saveDiff(
+  data,
+  callback = err => {
+    if (err) throw err;
+  }
+) {
+  fs.writeFile('diffBundles.txt', data, callback);
```
Comment (shaulgo) [inline plugins-bundle-analyzer/compareBundles.js:12]: if you use fs.writeFileSync you won't need this error callback. Change it or leave it, however, you like.

File: plugins-bundle-analyzer/analyzeBundles.js [hunk 6]
```diff
@@ -79,8 +88,11 @@ function run() {
         } else {
           const chlk = size > 500 ? warning : size > 250 ? chalk.yellow : chalk.green;
           console.log(prefix, chlk(`${size}KB`));
+          sizesObject[name] = size;
         }
       });
+
+      save(fileName, JSON.stringify(sizesObject));
```
Comment (shaulgo) [inline plugins-bundle-analyzer/analyzeBundles.js:95]: try this JSON.stringify(sizesObject, null, 2)

File: plugins-bundle-analyzer/updatePRComment.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+const fs = require('fs');
+
+const generateMessage = message => {
+  const titleForPRComment = `Significant differences between the bundle sizes:\n`;
+  return titleForPRComment.concat(message);
+};
+
+const updatePRComment = () => {
+  try {
+    const diff = fs.readFileSync('diffBundles.txt', 'utf8');
```
Comment (shaulgo) [inline plugins-bundle-analyzer/updatePRComment.js:12]: no need to save the diffBundles file.

File: package.json (hunk 2 above)
Comment (barboaron) [reply package.json:28]: Removed, I forgot to remove it

File: plugins-bundle-analyzer/analyzeBundles.js (hunk 6 above)
Comment (barboaron) [reply plugins-bundle-analyzer/analyzeBundles.js:95]: Done

File: plugins-bundle-analyzer/compareBundles.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+  console.log(chalk.magenta('compares bundle sizes to baseline...'));
+  Object.keys(currentBundles).forEach(key => {
+    const oldSize = savingBundles[key];
+    const newSize = currentBundles[key];
+    if (oldSize) {
+      if (newSize !== oldSize && parseInt(newSize) - parseInt(oldSize) > 10) {
```
Comment (shaulgo) [inline plugins-bundle-analyzer/compareBundles.js:28]:
```suggestion
      if (newSize !== oldSize && parseInt(newSize) - parseInt(oldSize) > 5) {
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -15,6 +15,9 @@
     Click to see more.
   </summary>
   
+### :house: Internal
+- `general`
+  - [#927](https://github.com/wix-incubator/rich-content/pull/927) Adding GitHub action that compares and fails if one of the current bundle sizes grows more then 10KB
```
Comment (shaulgo) [inline CHANGELOG.md:20]:
```suggestion
  - [#927](https://github.com/wix-incubator/rich-content/pull/927) Adding GitHub action that compares and fails if one of the current bundle sizes grows more then 5KB
```

**Pull request merged**